ttyrec: Use openpty from util.h

macOS ships openpty(3) in util.h, not libutil.h. Fix this by changing the
include.

Upstream-Status: Inappropriate [configuration]

--- ttyrec.c.orig	2020-06-17 01:25:44.000000000 +0200
+++ ttyrec.c	2020-06-17 01:25:55.000000000 +0200
@@ -71,7 +71,7 @@
 #define _(FOO) FOO
 
 #ifdef HAVE_openpty
-#include <libutil.h>
+#include <util.h>
 #endif
 
 #if defined(SVR4) && !defined(CDEL)
